sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOReadTest.java

```diff
@@ -906,4 +919,130 @@ public void testCoderInference() {
         KvCoder.of(ByteStringCoder.of(), ProtoCoder.of(Mutation.class)),
         BigQueryIO.read(parseFn).inferCoder(CoderRegistry.createDefault()));
   }
+
+  @Test
+  public void testReadFromParallelReadTableSource() throws IOException, InterruptedException {
```

Comment:
Please manually make sure that this test, when run under Coverage (e.g. you can do that in IntelliJ), covers near-100% of the new codepaths.

Reply:
Verified that the test coverage is near 100% for the new code paths.